world/console.cpp

```diff
@@ -962,6 +962,268 @@ void ConsoleQuit(
+		connection->SendLine("czcast expedition [expedition_id] [spell_id]");
+		connection->SendLine("czcast group [group_id] [spell_id]");
+		connection->SendLine("czcast guild [guild_id] [spell_id]");
+		connection->SendLine("czcast name [character_name] [spell_id]");
+		connection->SendLine("czcast raid [raid_id] [spell_id]");
+		return;
+	}
+
+	const auto& type = Strings::ToLower(args[0]);
+
+	const auto is_character  = type == "character";
+	const auto is_expedition = type == "expedition";
+	const auto is_group      = type == "group";
+	const auto is_guild      = type == "guild";
+	const auto is_name       = type == "name";
+	const auto is_raid       = type == "raid";
+
+	if (
+		!is_character &&
+		!is_expedition &&
+		!is_group &&
+		!is_guild &&
+		!is_name &
+		!is_raid
+	) {
+		connection->SendLine("czcast character [character_id] [spell_id]");
+		connection->SendLine("czcast expedition [expedition_id] [spell_id]");
+		connection->SendLine("czcast group [group_id] [spell_id]");
+		connection->SendLine("czcast guild [guild_id] [spell_id]");
+		connection->SendLine("czcast name [character_name] [spell_id]");
+		connection->SendLine("czcast raid [raid_id] [spell_id]");
+		return;
+	}
+
+	const auto   update_identifier = !is_name ? Strings::ToInt(args[1]) : 0;
+	const auto&  name              = is_name ? Strings::UcFirst(Strings::ToLower(args[1])) : std::string();
+	const auto   spell_id          = Strings::ToUnsignedInt(args[2]);
+
+	uint8 update_type;
+
+	if (is_character) {
+		update_type = CZUpdateType_Character;
+	} else if (is_expedition) {
+		update_type = CZUpdateType_Expedition;
+	} else if (is_group) {
+		update_type = CZUpdateType_Group;
+	} else if (is_guild) {
+		update_type = CZUpdateType_Guild;
+	} else if (is_name) {
+		update_type = CZUpdateType_ClientName;
+	} else if (is_raid) {
+		update_type = CZUpdateType_Raid;
+	}
+
+	auto pack = new ServerPacket(ServerOP_CZSpell, sizeof(CZSpell_Struct));
+	auto* CZS = (CZSpell_Struct*) pack->pBuffer;
+
+	CZS->update_type       = update_type;
+	CZS->update_subtype    = CZSpellUpdateSubtype_Cast;
+	CZS->update_identifier = update_identifier;
+	CZS->spell_id          = spell_id;
+
+	strn0cpy(CZS->client_name, name.c_str(), sizeof(CZS->client_name));
+
+	zoneserver_list.SendPacket(pack);
+	safe_delete(pack);
+
+	connection->SendLine(
+		fmt::format(
+			"Casting spell ID {} across zones by {} with an identifier of {}.",
+			spell_id,
+			type,
+			!is_name ? std::to_string(update_identifier) : name
+		)
+	);
+}
+
+void ConsoleWorldWideCastSpell(
+	EQ::Net::ConsoleServerConnection *connection,
+	const std::string &command,
+	const std::vector<std::string> &args
+) {
+	if (args.size() < 1) {
+		connection->SendLine("wwcast [spell_id]");
+		connection->SendLine("wwcast [spell_id] [min_status]");
+		connection->SendLine("wwcast [spell_id] [min_status] [max_status]");
+		return;
+	}
+
+	const auto spell_id = Strings::ToUnsignedInt(args[0]);
```

Comment:
We should cover some validation cases here

```
> wwcast 1 something 1
Casting spell ID 1 world wide for players with a status between 0 and 1.
> 
> 
> wwcast something 1 something
Casting spell ID 0 world wide for players with a status between 1 and 0.
> 
> 
> wwcast 1 something something
Casting spell ID 1 world wide for players with a status between 0 and 0.
```

Reply:
Not sure how to validate here, I've tried using `spells[spell_id]` or whatever, but it doesn't like that, also tried `GetSpellName(spell_id)` and it didn't like that.

Reply:
Tried adding a SharedDatabase method to get the spell's name from the database, but that didn't work either, just crashes world even on valid input.

Reply:
I meant moreso `string` vs `int` input validation for the different args. I don't think we need to load data into world to validate spells specifically